### DISCUSSION TITLE
What to do if installing extensions doesn't work?

### DISCUSSION THREAD

**Discussion by 1korn** [datenstrom/community#984] [Ask a question] [answered]
Hi,
I'm trying to use yellow as my CMS and the only extension I would need is gallery. But I can't get it to work.
I'm doing everything directly on a public webserver so not my environment. Installation works fine. My server is running PHP 8.2.
If I copy the extension to my extensions folder nothing happens. When I try to connect with SSH using the install command and I try the install command I get a 404.

The strange thing is that I get a 404 on every command, even update. I am in the right folder, I set the permissions for the whole folder to 755 so I really have no idea what I'm doing wrong.

Any suggestions?

**Comment by schulle4u** [comment]
Can you please describe step by step how you installed Yellow? I am pretty sure that the error does not occur on our side. I just tried a test installation on the command line and everything works as expected. 

```
steffen@Steffen-AMD:~$ git clone https://github.com/datenstrom/yellow.git yellowtest
Cloning into 'yellowtest'...
remote: Enumerating objects: 9217, done.
remote: Counting objects: 100% (1674/1674), done.
remote: Compressing objects: 100% (674/674), done.
remote: Total 9217 (delta 1112), reused 1494 (delta 974), pack-reused 7543
Receiving objects: 100% (9217/9217), 10.53 MiB | 8.37 MiB/s, done.
Resolving deltas: 100% (5861/5861), done.
steffen@Steffen-AMD:~$ cd yellowtest/
steffen@Steffen-AMD:~/yellowtest$ php yellow.php skip installation
Yellow skip: 11 extensions, 0 errors
steffen@Steffen-AMD:~/yellowtest$ php yellow.php install gallery
Yellow install: Website updated, 1 extension installed
steffen@Steffen-AMD:~/yellowtest$ 
```

Please note that this is not the usual installation procedure, I just wanted to see if extensions could be installed at all.

**Comment by annaesvensson** [comment]
What does the log say? You find the latest entries in file `system/extensions/yellow-website.log`.

**Comment by 1korn** [comment]
Thanks for responses.
It seems that my webserver is only responding with HTML to php commands so I guess that's one part of the problem.
The other part was me... I didn't get that you had to trigger update in the UI to install extensions.

It's working now, thanks

**Comment by annaesvensson** [comment, marked as answer]
It seems one step was missing. Open your website in a web browser and click "Reload".  See also datenstrom/community#920.

I added this to the [install instructions for extensions](https://github.com/annaesvensson/yellow-update). Hope this clarifies what needs to be done. In the long term we should probably simplify installing extensions, using FTP application or command line is not for everyone, adding support for installing extensions in web browser would make life easier for new users. I think it would also look beautiful in the UI.

PS: Welcome to our community. You can let us know when your question has been answered, click `Mark as answer`.
